src/Microsoft.AspNetCore.Cors/Infrastructure/CorsPolicy.cs

```diff
@@ -141,5 +149,23 @@ public override string ToString()
             builder.Append("}");
             return builder.ToString();
         }
+
+        /// <summary>
+        /// Creates a new instance of the <see cref="CorsPolicy"/>.
+        /// </summary>
+        public CorsPolicy()
+        {
+            // Regenerate 'RegularOrigins' every time the 'Origins' has changed.
+            Origins.CollectionChanged += (sender, e) =>
```

Comment:
I also don't see how this is related to the PR. Can you make this change only about wildcards and separate dynamic configuration in another issue/PR.